htdocs/PI/index.php

```diff
@@ -367,11 +370,48 @@ function getXml() {
 
     /* Authorize a user based on their certificate */
 
-    function authAnyCert() {
-        if (empty($this->dn))
-            die("<No valid certificate found. A trusted certificate is " .
+    function authByCert() {
+        require_once __DIR__.'/../web_portal/controllers/utils.php';
+        require_once __DIR__.'/../../lib/Doctrine/entities/APIAuthentication.php';
+
+        if (empty($this->dn)) {
+            throw new \Exception("No valid certificate found. A trusted certificate is " .
                     "required to access this resource. Try accessing the " .
                     "resource through the private interface.");
+        }
+
+        $admin = false;
+        $authenticated = false;
+
+        $user = \Factory::getUserService()->getUserByPrinciple($this->dn);
+
+        if ($user == null) {
+            // Incoming credential is not that of a registered user.
+            // Check if it is registered API Authentication credential.
+
+            $authEntServ = \Factory::getAPIAuthenticationService();
+            $authEnt = $authEntServ->getAPIAuthentication($this->dn, "X509");
```

Comment:
with `"X509"` being hard coded here, what happens if `$this->dn` isn't a DN? What if it's a EGI Check-In identifier?

Reply:
The honest answer is that I don't know, but since X509 is hard coded in the available types on instance creation, it seems the safest option for now. Has the identification of an EGI identifier progressed more outside of the dev branch? https://github.com/GOCDB/gocdb/blob/facc3167e2b28be0524ddeaa3241edbdcfe12008/htdocs/web_portal/controllers/site/add_api_auth.php#L61

Reply:
> Has the identification of an EGI identifier progressed more outside of the dev branch?

Sarah's work will add "OIDC Subject", so I think there will be some interplay here. Hopefully not too much as the `APIAuthenticationService` looks pretty "type" agnostic to me :).

Reply:
Let's hope so. There doesn't seem much that can be done regarding this PR?

Reply:
Out of interest: why 'OIDC Subject' rather than just 'OIDC', which seems more in line with 'X509' rather than 'X509 Subject'? Do we expect 'OIDC Somethingelse'?

Reply:
> Let's hope so. There doesn't seem much that can be done regarding this PR?

No yet 😄 

> Out of interest: why 'OIDC Subject' rather than just 'OIDC', which seems more in line with 'X509' rather than 'X509 Subject'? Do we expect 'OIDC Somethingelse'?

We should probably take this over to #260 😋